SDR tests are failing with gdal=3.4.2

Three SDR tests are failing on github actions:
```
test_base_regression: AssertionError: these key (actual/expected) errors occured: {'usle_tot': (13.90210914612, 14.25030517578), 'sed_export': (0.55185163021, 0.60502111912), 'sed_dep': (8.80612564087, 9.05251502991)}

test_drainage_regression: AssertionError: these key (actual/expected) errors occured: {'sed_export': (0.97192692757, 1.02959537506), 'usle_tot': (12.68887424469, 12.97211265564)}

test_regression_with_undefined_nodata: AssertionError: these key (actual/expected) errors occured: {'sed_export': (0.55185163021, 0.60502111912), 'usle_tot': (13.90210914612, 14.25030517578)}
```
These passed when I ran them locally. Possibly an updated dependency?

--- a/HISTORY.rst
+++ b/HISTORY.rst
@@ -39,6 +39,8 @@ Unreleased Changes
 * General
     * Update python packaging settings to exclude a few config files and the
       workbench from source distributions and wheels
+    * Updating SDR test values due to an update in GDAL's mode resampling
+      algorithm. See https://github.com/natcap/invest/issues/905
 * Workbench
     * Fixed a bug where some model runs would not generate a new item
       in the list of recent runs.

--- a/requirements-docs.yml
+++ b/requirements-docs.yml
@@ -12,7 +12,7 @@ channels:
 dependencies:
 - chardet>=3.0.4
 - Cython
-- GDAL>=3.1.2,!=3.3.0,<3.4.1
+- GDAL>=3.4.2
 - Flask
 - flask_cors
 - numpy>=1.11.0,!=1.16.0

--- a/requirements.txt
+++ b/requirements.txt
@@ -12,7 +12,7 @@
 # scripts/convert-requirements-to-conda-yml.py as though it can only be found
 # on pip.
 
-GDAL>=3.1.2,!=3.3.0
+GDAL>=3.4.2
 Pyro4==4.77  # pip-only
 pandas>=1.2.1
 numpy>=1.11.0,!=1.16.0

--- a/requirements.txt
+++ b/requirements.txt
@@ -12,7 +12,7 @@
 # scripts/convert-requirements-to-conda-yml.py as though it can only be found
 # on pip.
 
-GDAL>=3.1.2,!=3.3.0,<3.4.1  # https://github.com/OSGeo/gdal/issues/3898 https://github.com/natcap/invest/issues/905
+GDAL>=3.1.2,!=3.3.0
 Pyro4==4.77  # pip-only
 pandas>=1.2.1
 numpy>=1.11.0,!=1.16.0

--- a/tests/test_sdr.py
+++ b/tests/test_sdr.py
@@ -214,10 +214,10 @@ class SDRTests(unittest.TestCase):
 
         sdr.execute(args)
         expected_results = {
-            'usle_tot': 14.25030517578,
+            'usle_tot': 13.90210914612,
             'sed_retent': 308382.125,
-            'sed_export': 0.60502111912,
-            'sed_dep': 9.05251502991
+            'sed_export': 0.55185163021,
+            'sed_dep': 8.80612564087
         }
 
         vector_path = os.path.join(
@@ -286,8 +286,8 @@ class SDRTests(unittest.TestCase):
         sdr.execute(args)
         expected_results = {
             'sed_retent': 308382.125,
-            'sed_export': 0.60502111912,
-            'usle_tot': 14.25030517578,
+            'sed_export': 0.55185163021,
+            'usle_tot': 13.90210914612,
         }
 
         vector_path = os.path.join(
@@ -335,8 +335,8 @@ class SDRTests(unittest.TestCase):
 
         expected_results = {
             'sed_retent': 476649.875,
-            'sed_export': 1.02959537506,
-            'usle_tot': 12.97211265564,
+            'sed_export': 0.97192692757,
+            'usle_tot': 12.68887424469,
         }
 
         vector_path = os.path.join(

